src/hooks/useApproveCallback.ts

```diff
@@ -134,19 +131,15 @@ export function useApproveCallbackFromTrade(trade?: Trade /* allowedSlippage = 0
 
   // Find the approve address for the trade
   let approveAddress = AddressZero
-  if (
-    trade instanceof CurveTrade ||
-    trade instanceof CoWTrade ||
-    trade instanceof UniswapTrade ||
-    trade instanceof ZeroXTrade
-  ) {
-    approveAddress = trade.approveAddress
-  } else if (trade instanceof UniswapV2Trade) {
+
+  if (trade instanceof UniswapV2Trade) {
     /**
      * @todo use approveAddress property in next version
      */
     const routerAddressList = trade.platform as UniswapV2RoutablePlatform
     approveAddress = routerAddressList.routerAddress[chainId as ChainId] as string
+  } else if (trade) {
+    approveAddress = trade.approveAddress
   }
```

Comment:
Nice cleanup. We should add approveAddress to UnsiwapV2Trade too

Reply:
https://github.com/levelkdev/swapr-sdk/issues/234 made an issue